### ISSUE TITLE
boxplot() return type

### ISSUE THREAD

**Issue by cachitas** [mwaskom/seaborn#623]
Is it possible to access the boxplot to further modify it?
Currently boxplot() returns the Axes. 

I would like to paint the boxes differently along `x` while having an hue coded with a pattern, and I'm trying to achieve this hacking around... ([question here](http://stackoverflow.com/questions/31206474/boxplot-colors-by-x-value-and-hues-differentiated-by-pattern))

Pandas addresses this by having an argument return_type ([see here](http://pandas.pydata.org/pandas-docs/stable/visualization.html#box-plots)).

**Comment by mwaskom** [comment]
You can pass `{artist}props` dictionaries to `boxplot` which are applied after the seaborn attributes. Also, references to the various artists are stored on the Axes object, you just have to know where to look. Then you can work with them directly.

The idea that Axes-level functions in seaborn return the Axes the plot is on is fundamental and will not change.

**Comment by cachitas** [comment]
I understand your point.
It is a `matplotlib` 'problem' then, to know where to look. I have played with the `get_lines()` but it is a bit messy. Also thank you for your answer on SO: `ax.artists` gives me access to the boxes indeed. I am guessing that the lines will give me the whiskers and the caps, but it is not very clear.
Anyway, thank you for the directions!

Meanwhile, I have managed to get the desired plot by using `matplotlib` only... No `seaborn` love involved but it got the job done.
